### PR TITLE
Make rpy2 and anndata2ri optional

### PR DESCRIPTION
**Pull request by mumichae**

### CODE REVIEW

File: scib/metrics/lisi.py [hunk 1]
```diff
@@ -188,6 +180,13 @@ def lisi_graph_py(
     Compute LISI score on shortes path based on kNN graph provided in the adata object.
     By default, perplexity is chosen as 1/3 * number of nearest neighbours in the knn-graph.
     """
+    try:
+        import rpy2.rinterface_lib.callbacks
+        import rpy2.rinterface_lib.embedded
+
+        rpy2.rinterface_lib.callbacks.logger.setLevel(logging.ERROR)
+    except ModuleNotFoundError as e:
+        raise OptionalDependencyNotInstalled(e)
```
Comment (LuckyMD) [inline scib/metrics/lisi.py:189]: I don't think this is needed here. the only rpy2 command is an RRuntimeError, which is actually not needed anymore as it's all computed in python and C++, no? Need to check with @mbuttner, but this looks like legacy rpy2 code to me and so rpy2 isn't needed for lisi computation at all.

**Review by LuckyMD — CHANGES_REQUESTED** [review]
Two things:
1. I think rpy2 isn't actually needed for lisi as we only use python and C++ and the only rpy2 parts are legacy code I think (in the lisi_graph function that is actually used and not deprecated) -> check this with @mbuttner 
2. If we make rpy2 optional, we also need to adapt the scib-pipeline install commands.

File: scib/metrics/lisi.py (hunk 1 above)
Comment (mumichae) [reply scib/metrics/lisi.py:189]: True, we wouldn't expect an R Runtime error. Do we still want to keep the behaviour of LISI returning NaN when the kNN graph fails?

Comment (mbuttner) [reply scib/metrics/lisi.py:189]: LISI does not need any rpy2 anymore. As @LuckyMD said, it is all Python and C++. The old LISI implementation needed R.

Comment (LuckyMD) [reply scib/metrics/lisi.py:189]: thanks for confirmation, @mbuttner ! And @mumichae: yes, NaN when kNN fails is good.

Comment (mumichae) [reply scib/metrics/lisi.py:189]: OK, I'm catching a `RuntimeError` now instead

Comment (LuckyMD) [reply scib/metrics/lisi.py]: Cool!

**Comment by LuckyMD** [comment]
Before mergeable then only need:
 - [] test with different env setups
 - [] scib-pipeline PR for install scripts

**Comment by mumichae** [comment]
For scib-pipeline, I don't think we need to change anything, since rpy2 is added as a dependency in the conda environments, regardless of the scib dependencies

**Comment by mumichae** [comment]
scib-pipeline CI is running through using the dependencies https://github.com/theislab/scib-pipeline/pull/45

File: scib/preprocessing.py [hunk 2]
```diff
@@ -774,7 +788,9 @@ def read_conos(inPath, dir_path=None):
     """
     from shutil import rmtree
 
-    import pandas as pd
+    import rpy2.rinterface_lib.callbacks
+    import rpy2.rinterface_lib.embedded
+    import rpy2.robjects as ro
```
Comment (LuckyMD) [inline scib/preprocessing.py:793]: why are the rpy2 import statements not wrapped in a try-catch block in this file?

**Review by LuckyMD — APPROVED** [review]
looks good to me, but just 1 question about wrapping rpy2 imports in the preprocessing file.

**Pull request merged**